tests/unit/reportGen.unitTest.ts

```diff
@@ -186,15 +189,108 @@ describe("report-gen", () => {
                     stream.push(null);
 
                     return workbook.xlsx.read(stream)
-                    .then((excelFile: any) => {
+                    .then((excelFile: Excel.Workbook) => {
                         const reportSheet: Excel.Worksheet = excelFile.getWorksheet(1);
 
                         expect(excelFile.creator).to.equal("Commercial Vehicles Services Beta Team");
+                        // @ts-ignore
                         expect(excelFile.company).to.equal("Drivers and Vehicles Standards Agency");
                         expect(reportSheet.name).to.equal("Retrokey report");
                     });
                 });
             });
+
+            context("the report contains prohibitionIssued false on testType level and true on defects level", () => {
+                it("should contain on the corresponding testType line, on the failureAdvisoryItemsQAICommentsTestValue column, the info that the prohibition WAS" +
+                    " issued on defects level and none on the Additional test type notes level", () => {
+                    return retroGenerationService.generateRetroReport(activity)
+                        .then((result: any) => {
+                            const workbook = new Excel.Workbook();
+                            const stream = new Duplex();
+                            stream.push(result.fileBuffer);
+                            stream.push(null);
+
+                            return workbook.xlsx.read(stream)
+                                .then((excelFile: Excel.Workbook) => {
+                                    const reportSheet: Excel.Worksheet = excelFile.getWorksheet(1);
+                                    const failureAdvisoryItemsQAICommentsTestValue = reportSheet.getCell("M17").value;
+                                    expect (failureAdvisoryItemsQAICommentsTestValue).to.contain("Prohibition was issued");
+                                    expect (failureAdvisoryItemsQAICommentsTestValue).to.contain("Additional test type notes: none;");
+                                });
+                        });
+                });
+
+            });
+
+
+            context("the report contains prohibitionIssued false on testType level and false on defects level", () => {
+                it("should contain on the corresponding testType line, on the failureAdvisoryItemsQAICommentsTestValue column, the info that the prohibition WAS" +
+                    " issued on defects level and that the prohibition WAS also issued on the Additional test type notes level", () => {
```

Comment:
Guess the descriptions are not in sync. Will need to be changed to Not issued and none for additional notes.

Reply:
fixed